script/hassfest/serializer.py

```diff
@@ -104,4 +105,6 @@ def _get_annotation(key: str) -> str:
     if annotations:
         # If we had any annotations, add the __future__ import.
         code = f"from __future__ import annotations\n{code}"
+    if typing_imports:
+        code = f"from typing import {','.join(sorted(typing_imports))}\n{code}"
```

Comment:
I think that all files genereated via hassfest are constants, so I'm not convinced it needs an extra argument.

Would it not make more sense to always add "Final"?
```suggestion
        code = f"from __future__ import annotations\n\nfrom typing import Final\n{code}"
```

```python
    return format_python_namespace(
        {"DHCP": match_list},
        annotations={"DHCP": "Final[list[dict[str, str | bool]]]"},
    )
...
    return format_python_namespace(
        {"BLUETOOTH": match_list},
        annotations={
            "BLUETOOTH": "Final[list[dict[str, bool | str | int | list[int]]]]"
        },
    )
```

Reply:
I agree it makes sense to always add `Final`. We would also need a `#noqa` for unused imports until all existing files are fixed.

Should I also change all existing generated files in this PR?

Reply:
`#noqa` should not be necessary, if you adjust the annotations of `dhcp` and `bluetooth`
The other files currently don't have annotations, so no need to update them.

Reply:
Fixed now, but I cannot see why hassfest isn't working - seems to be a import sort issue.

Reply:
You didn't use my suggestion above.
```python
code = f"from __future__ import annotations\n\nfrom typing import Final\n{code}"
```